serde/src/java/org/apache/hadoop/hive/serde2/avro/InstanceCache.java

```diff
@@ -51,20 +51,19 @@ public Instance retrieve(SeedObject hv) throws AvroSerdeException {
    * Retrieve (or create if it doesn't exist) the correct Instance for this
    * SeedObject using 'seenSchemas' to resolve circular references
    */
-  public synchronized Instance retrieve(SeedObject hv,
-      Set<SeedObject> seenSchemas) throws AvroSerdeException {
+  public Instance retrieve(SeedObject hv, Set<SeedObject> seenSchemas)
+    throws AvroSerdeException {
     if(LOG.isDebugEnabled()) LOG.debug("Checking for hv: " + hv.toString());
```

Comment:
Please change this (and the rest) to:
`
LOG.debug("Checking for hv: {}", hv);
`
http://www.slf4j.org/faq.html#logging_performance